obspy/signal/interpolation.py

```diff
@@ -172,6 +173,255 @@ def weighted_average_slopes(data, old_start, old_dt, new_start, new_dt,
     return return_data
 
 
+# Map corresponding to the enum on the C side of things.
+_LANCZOS_KERNEL_MAP = {
+    "lanczos": 0,
+    "hanning": 1,
+    "blackman": 2
+}
+
+
+def lanczos_interpolation(data, old_start, old_dt, new_start, new_dt, new_npts,
+                          a, window="lanczos", *args, **kwargs):
+    r"""
+    Function performing Lanczos resampling, see
+    http://en.wikipedia.org/wiki/Lanczos_resampling for details. Essentially a
+    finite support version of sinc resampling (the ideal reconstruction
+    filter). For large values of ``a`` it converges towards sinc resampling. If
+    used for downsampling, make sure to apply a proper anti-aliasing lowpass
+    filter first.
+
+    .. note::
+
+        In most cases you do not want to call this method directly but invoke
+        it via either the :meth:`obspy.core.stream.Stream.interpolate` or the
+        :meth:`obspy.core.trace.Trace.interpolate` method. These offer a nicer
+        API that naturally integrates with the rest of ObsPy. Use
+        ``method="lanczos"`` to use this interpolation method. In that case the
+        only additional parameters of interest are ``a`` and ``window``.
+
+    :type data: array_like
+    :param data: Array to interpolate.
+    :type old_start: float
+    :param old_start: The start of the array as a number.
+    :type old_start: float
+    :param old_dt: The time delta of the current array.
+    :type new_start: float
+    :param new_start: The start of the interpolated array. Must be greater
+        or equal to the current start of the array.
+    :type new_dt: float
+    :param new_dt: The desired new time delta.
+    :type new_npts: int
+    :param new_npts: The new number of samples.
+    :type a: int
+    :param a: The width of the window in samples on either side. Runtimes
+        scales linearly with the value of ``a`` but the interpolation also get
+        better.
+    :type window: str
+    :param window: The window used to multiply the sinc function with. One
```

Comment:
The `with` is a bit strange here. But you can replace `multiple` with `taper` and drop the `with`, right?